arch/arm/src/arm/Toolchain.defs

```diff
@@ -67,6 +67,10 @@ else
   MAXOPTIMIZATION ?= -Os
 endif
 
+ifeq ($(CONFIG_MM_KASAN),y)
+  MAXOPTIMIZATION += -fsanitize=kernel-address
```

Comment:
Adding it as an "optimization option" seems conceptually wrong. We should create another variable for the instrumentation options.

Reply:
The new option require we modify all boards/**/scripts/Make.defs, or we let user add -fsanitize=kernel-address manually.

Reply:
I agree with Gustavo here.  We are kind of abusing the MAXOPTIMIZATION variable here.

Reply:
Ok, I revert all change under arch/, user has to add -fsanitize=kernel-address.